spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/TestSparkV2Filters.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.Expressions;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.spark.sql.connector.expressions.FieldReference;
+import org.apache.spark.sql.connector.expressions.LiteralValue;
+import org.apache.spark.sql.connector.expressions.NamedReference;
+import org.apache.spark.sql.connector.expressions.filter.And;
+import org.apache.spark.sql.connector.expressions.filter.Not;
+import org.apache.spark.sql.connector.expressions.filter.Or;
+import org.apache.spark.sql.connector.expressions.filter.Predicate;
+import org.apache.spark.sql.types.DateType$;
+import org.apache.spark.sql.types.IntegerType$;
+import org.apache.spark.sql.types.TimestampType$;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestSparkV2Filters {
```

Comment:
Do we need some test of the cases of the expression the other way ( literal, attribute)

Reply:
Added some tests. Thanks!

Reply:
Nit / non-blocking / for follow up if desired: We might break these up into two or three sets of tests.

I can see:
1. Literal to Attribute
2. Attribute to Literal
3. Invalid assertions.

This could totally be done in a follow up PR, but because the stringly typed nature of these areas is somewhat error prone in my opinion I think we should try to make the tests extensible. But not important for this PR imo (which has been around long enough and would be good to be using).

Reply:
I will refactor the tests in a follow up. Thanks for the suggestion!